File: src/SIStatisticsService.Contract/Models/GameResultInfo.cs
namespace SIStatisticsService.Contract.Models;

/// <summary>
/// Defines game result info.
/// </summary>
/// <param name="Package">Game package info.</param>
/// <param name="LanguageCode">Game language code.</param>
public sealed record GameResultInfo(PackageInfo Package, string? LanguageCode = null)
{
    /// <summary>
    /// Game name.
    /// </summary>
    public string Name { get; set; } = "";

    /// <summary>
    /// Game platform.
    /// </summary>
    public GamePlatforms Platform { get; set; }

    /// <summary>
    /// Game finish time.
    /// </summary>
    public DateTimeOffset FinishTime { get; set; }

    /// <summary>
    /// Game duration.
    /// </summary>
    public TimeSpan Duration { get; set; } = TimeSpan.Zero;

    /// <summary>
    /// Game results: player names and their scores.
    /// </summary>
    public Dictionary<string, int> Results { get; init; } = new();

    /// <summary>
    /// Player reviews.
    /// </summary>
    public Dictionary<string, string> Reviews { get; init; } = new();
}

File: web/src/models/GameResultInfo.ts
import GamePlatforms from "./GamePlatforms";
import PackageInfo from "./PackageInfo";

/** Defines game result info. */
export default interface GameResultInfo {
	/** Game name. */
	name: string;

	/** Game platform. */
	platform: GamePlatforms;

	/** Game finish time. */
	finishTime: Date;

	/** Game duration. */
	duration: string;

	/** Game package info. */
	package: PackageInfo;

	/** Game results: player names and their scores. */
	results: Record<string, number>;

	/** Player reviews. */
	reviews: Record<string, number>;
}
